mempool/src/core_mempool/mempool.rs

```diff
@@ -178,8 +199,12 @@ impl Mempool {
             let account_sequence_number = self.sequence_number_cache.get(&txn.address);
             let seen_previous = seq > 0 && seen.contains(&(txn.address, seq - 1));
             // include transaction if it's "next" for given account or
-            // we've already sent its ancestor to Consensus
-            if seen_previous || account_sequence_number == Some(&seq) {
+            // we've already sent its ancestor to Consensus. In the case of CRSNs, we can safely
+            // assume that it can be included.
```

Comment:
because the code for including the tx in the mempool in the first place already checks that the CRSN is < the RHS of the window?

Reply:
Correct! The transaction won't be admitted to mempool unless its sequence nonce is valid (so within the window). As you pointed out, it could be no longer valid under force-shifting though, however there isn't an easy way of getting this information I believe on commit :(